File: BomPriceApproval.API/Domain/Entities/CompanySettings.cs
namespace BomPriceApproval.API.Domain.Entities;

public class CompanySettings
{
    public int Id { get; set; }                          // always 1 (singleton)
    public string CompanyName { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string Telephone { get; set; } = string.Empty;
    public string Trn { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Website { get; set; } = string.Empty;
    public int QuotationValidityDays { get; set; } = 30;
    public string TermsAndConditions { get; set; } = string.Empty;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public int? UpdatedByUserId { get; set; }
    public User? UpdatedByUser { get; set; }
}

File: bom-web/src/api/companySettings.ts
import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
import { api } from "@/api/axios";

export interface CompanySettings {
  companyName: string;
  address: string;
  telephone: string;
  trn: string;
  email: string;
  website: string;
  quotationValidityDays: number;
  termsAndConditions: string;
  updatedAt: string;
  updatedByName: string | null;
}

export interface UpdateCompanySettingsRequest {
  companyName: string;
  address: string;
  telephone: string;
  trn: string;
  email: string;
  website: string;
  quotationValidityDays: number;
  termsAndConditions: string;
  reason: string;
}

const QUERY_KEY = ["admin", "company-settings"] as const;

export function useCompanySettings() {
  return useQuery({
    queryKey: QUERY_KEY,
    queryFn: async () => {
      const { data } = await api.get<CompanySettings>("/admin/company-settings");
      return data;
    },
  });
}

export function useUpdateCompanySettings() {
  const qc = useQueryClient();
  return useMutation({
    mutationFn: async (body: UpdateCompanySettingsRequest) => {
      const { data } = await api.put<CompanySettings>("/admin/company-settings", body);
      return data;
    },
    onSuccess: (data) => {
      qc.setQueryData(QUERY_KEY, data);
    },
  });
}
